origin/metrics.go

```diff
@@ -28,162 +30,162 @@ type muxerMetrics struct {
 }
 
 type TunnelMetrics struct {
-	haConnections     prometheus.Gauge
-	totalRequests     prometheus.Counter
-	requestsPerTunnel *prometheus.CounterVec
+	haConnections prometheus.Gauge
+
+	requests *prometheus.CounterVec
 	// concurrentRequestsLock is a mutex for concurrentRequests and maxConcurrentRequests
 	concurrentRequestsLock      sync.Mutex
 	concurrentRequestsPerTunnel *prometheus.GaugeVec
-	// concurrentRequests records count of concurrent requests for each tunnel
-	concurrentRequests             map[string]uint64
+	// concurrentRequests records count of concurrent requests for each tunnel, keyed by hash of label values
+	concurrentRequests             map[uint64]uint64
 	maxConcurrentRequestsPerTunnel *prometheus.GaugeVec
-	// concurrentRequests records max count of concurrent requests for each tunnel
-	maxConcurrentRequests map[string]uint64
+	// concurrentRequests records max count of concurrent requests for each tunnel, keyed by hash of label values
+	maxConcurrentRequests map[uint64]uint64
 	timerRetries          prometheus.Gauge
-	responseByCode        *prometheus.CounterVec
-	responseCodePerTunnel *prometheus.CounterVec
-	serverLocations       *prometheus.GaugeVec
+
+	responses       *prometheus.CounterVec
+	serverLocations *prometheus.GaugeVec
 	// locationLock is a mutex for oldServerLocations
 	locationLock sync.Mutex
 	// oldServerLocations stores the last server the tunnel was connected to
-	oldServerLocations map[string]string
+	oldServerLocations map[uint64]string
 
 	muxerMetrics *muxerMetrics
 }
 
-func newMuxerMetrics() *muxerMetrics {
+func newMuxerMetrics(baseMetricsLabels []string) *muxerMetrics {
 	rtt := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: "rtt",
+			Name: "argotunnel_rtt",
```

Comment:
We should make argotunnel the namespace of all the metrics in this file.

Reply:
https://godoc.org/github.com/prometheus/client_golang/prometheus#Opts has `Namespace` -- we should use this instead (@chungthuang is right)